### ISSUE TITLE
google_compute_instance_group_manager is deleted when operation is DONE then expires

### ISSUE THREAD

**Issue by dlott** [hashicorp/terraform-provider-google#16043]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform v0.13.7
+ provider registry.terraform.io/hashicorp/google v3.46.0
+ provider registry.terraform.io/hashicorp/google-beta v4.75.0

### Affected Resource(s)

* google_compute_instance_group_manager

### Terraform State Files

excerpt of .tfstate
```tf
    {
      "type": "google_compute_instance_group_manager",
      "name": "app_mig",
      "provider": "provider[\"registry.terraform.io/hashicorp/google-beta\"]",
      "instances": [
        {
            "operation": "operation-1694571844655-6053440d8d7e2-44f1c718-fec0049a",
```

Note that this operation is now complete but still persisted in state file for unknown reasons (probably timeout/interruption during an apply)

### Debug Output

For two weeks, other applies make changes to other resources, but leave this unchanged.

```
2023-09-22T16:24:41.491-0700 [DEBUG] plugin.terraform-provider-google-beta: 2023/09/22 16:24:41 [DEBUG] in progress operation detected at operation-1694567561052-6053341864268-1d34a7ba-07262f4c, attempting to resume
2023-09-22T16:24:42.472-0700 [DEBUG] plugin.terraform-provider-google-beta: 2023/09/22 16:24:42 [DEBUG] Got DONE while polling for operation operation-1694567561052-6053341864268-1d34a7ba-07262f4c's status
```

But after 2 weeks, the operation history is expires and the operation returns `404`

```
2023-09-27T10:25:19.574-0700 [DEBUG] plugin.terraform-provider-google-beta: 2023/09/27 10:25:19 [DEBUG] Dismissed an error as retryable. Retry 404s for GET operation - googleapi: Error 404: The resource 'projects/p3rf-chasm-app-deep/zones/asia-southeast1-b/operations/operation-1694571844655-6053440d8d7e2-44f1c718-fec0049a' was not found, notFound
2023-09-27T10:25:19.574-0700 [DEBUG] plugin.terraform-provider-google-beta: 2023/09/27 10:25:19 [DEBUG] Dismissed retryable error on GET operation "operation-1694571844655-6053440d8d7e2-44f1c718-fec0049a": googleapi: Error 404: The resource 'projects/p3rf-chasm-app-deep/zones/asia-southeast1-b/operations/operation-1694571844655-6053440d8d7e2-44f1c718-fec0049a' was not found, notFound
2023-09-27T10:25:19.574-0700 [DEBUG] plugin.terraform-provider-google-beta: 2023/09/27 10:25:19 [DEBUG] Resumed operation returned an error, removing from state: Error waiting for Creating InstanceGroupManager: couldn't find resource (21 retries)
module.gcp_deep_asia-southeast1-b.module.deep_app[0].module.app_load_balancer.google_compute_backend_service.app_backend: Refreshing state... [id=projects/p3rf-chasm-app-deep/global/backendServices/deep-prod-asia-southeast1-b-be]
```

The state is removed from the state file, per https://github.com/hashicorp/terraform-provider-google-beta/blob/2fbb5de151d0831d244f9bbc9ff2c116c006937c/google-beta/services/compute/resource_compute_instance_group_manager.go#L715
 even though the resource continues to exist.

Further `terraform apply` attempts fail due to duplicate name issues.

### Expected Behavior

Once the provider saw the operation was `DONE` it should have set `"operation": null` in the state file so that it is not removed when the operation history expires.

b/312433487

**Comment by edwardmedia** [comment]
@dlott can you share your debug log so I can see what is going on?
 Also, please share you config and detailed steps so I can repro?

**Comment by edwardmedia** [comment]
@hao-nan-li you probably can repro it by turning your machine off when `google_compute_instance_group_manager` is still applying. At that time, the operation id is still captured in the state. If you reapply right away, it probably will pick the process up and continue to finish without hitting a problem. But it looks like the api keeps the operation id for 2 weeks, if you pick up the process after two weeks, you may hit this problem.

**Comment by hao-nan-li** [comment]
Did you perform any other operations that caused the deletion of that resource?

**Comment by melinath** [comment]
I was able to reproduce this issue by interrupting `terraform apply` with Ctrl + C. It looks like that causes the initial create to store the operation ID in the state file:

https://github.com/hashicorp/terraform-provider-google-beta/blob/8f1823692074b62582d1318e1d571d0f86d200ae/google-beta/services/compute/resource_compute_instance_group_manager.go#L604-L606

If the operation ID is set, it gets persisted indefinitely and re-checked on every read.

https://github.com/hashicorp/terraform-provider-google-beta/blob/8f1823692074b62582d1318e1d571d0f86d200ae/google-beta/services/compute/resource_compute_instance_group_manager.go#L706-L724

I can't easily verify the 2-week issue but there's definitely no code I see to prevent the operation from being persisted (or to unset it) so this seems like a plausible issue.

For comparison, `google_container_cluster` also has this "operation"-persisting behavior but its Read function always _unsets_ the operation:

https://github.com/hashicorp/terraform-provider-google-beta/blob/8f1823692074b62582d1318e1d571d0f86d200ae/google-beta/services/container/resource_container_cluster.go#L2531-L2539

It looks like google_compute_instance_group_manager originally also always unset the operation but was accidentally changed to this behavior as part of a large lint-fixing change (probably a copy-paste error): https://github.com/hashicorp/terraform-provider-google-beta/commit/79e0a2e37e138ec8ade75ac726e3de4dc9adfc88#diff-1cf6fdb65a48539ac4730b8920d0366d3a173b7e824b08502d3c3da73456d8f8L466

So in conclusion - the fix here is likely that Read() should always unset the operation field.
